sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -558,26 +558,51 @@ private static DDlogRecord toValue(final Field<?> field, final Object in) {
         }
     }
 
-    private static void setValue(final Field<?> field, final DDlogRecord in, final Record out) {
-        final DataType<?> dataType = field.getDataType();
-        switch (dataType.getSQLType()) {
-            case Types.BOOLEAN:
-                out.setValue((Field<Boolean>) field, in.getBoolean());
-                return;
-            case Types.INTEGER:
-                out.setValue((Field<Integer>) field, in.getInt().intValue());
-                return;
-            case Types.BIGINT:
-                out.setValue((Field<Long>) field, in.getInt().longValue());
-                return;
-            case Types.VARCHAR:
-                out.setValue((Field<String>) field, in.getString());
-                return;
-            default:
-                throw new DDlogJooqProviderException("Unknown datatype " + field);
+    // Return a value based on the DDlog type of the record
+    private static Object ddlogRecordToObject(final Field<?> field, final DDlogRecord in) {
+        if (in.isBool()) {
```

Comment:
How about adding a 'debug' flag which will cause the code to assert that the SQL type matches as well?